### ISSUE TITLE
SAINT pre-processing (Galaxy Version 1.0.0) does not recognize my file

### ISSUE THREAD

**Issue by mavino** [bornea/APOSTL#6]
I have a text file which is supposed to be my maxquant peptide.txt file (I say supposed because I have not personally run the maxquant analysis, but I was given the output file). It is tsv file and contains these kind of columns:
```
Protein IDs	Majority protein IDs	Peptide counts (all)	Peptide counts (razor+unique)	Peptide counts (unique)	Fasta headers	Number of proteins	Peptides	Razor + unique peptides	Unique peptides	Peptides strep str3 BioID delta IP HEDE (2) ??	Peptides strep str3 BioID Fe (3) ??	Peptides strep str3 delta DIP HEPE (5) ??	Razor + unique peptides strep str3 BioID delta IP HEDE (2) ??	Razor + unique peptides strep str3 BioID Fe (3) ??	Razor + unique peptides strep str3 delta DIP HEPE (5) ??	Unique peptides strep str3 BioID delta IP HEDE (2) ??	Unique peptides strep str3 BioID Fe (3) ??	Unique peptides strep str3 delta DIP HEPE (5) ??	Sequence coverage [%]	Unique + razor sequence coverage [%]	Unique sequence coverage [%]	Mol. weight [kDa]	Sequence length	Sequence lengths	Q-value	Score	Sequence coverage strep str3 BioID delta IP HEDE (2) ?? [%]	Sequence coverage strep str3 BioID Fe (3) ?? [%]	Sequence coverage strep str3 delta DIP HEPE (5) ?? [%]	Intensity	Intensity strep str3 BioID IP HEME (2) ??	Intensity strep str3 BioID Fe (3) ??	Intensity strep str3 delta DIP HEME (5) ??	MS/MS Count	Only identified by site	Reverse	Potential contaminant	id	Peptide IDs	Peptide is razor	Mod. peptide IDs	Evidence IDs	MS/MS IDs	Best MS/MS	Oxidation (M) site IDs	Oxidation (M) site positions
```
The problem is that SAINT pre-processing on my locally installed Galaxy does not recognize this file.
Thank you very much.

**Comment by pstew** [comment]
Hello, thanks for using the tool. Try uploading the file and specifying a `.txt` file type. Sometimes Galaxy can auto-detect the wrong thing and this can cause an issue. Also, try removing the spaces in the sample names (and then make sure this matches in your other files). The MaxQuant file gets read into an R instance in the background, and R autoconverts spaces to `.`, which can cause an issue when attempting to match `strep.str3` to `strep str3`.

**Comment by mavino** [comment]
Thanks for replying...these are my column names now:
```
Protein IDs	Majority protein IDs	Peptide counts (all)	Peptide counts (razor+unique)	Peptide counts (unique)	Fasta headers	Number of proteins	Peptides	Razor + unique peptides	Unique peptides	Peptides strepstr3BioIDdeltaIPHEDE2	Peptides strepstr3BioIDFe3	Peptides strepstr3deltaDIPHEPE5	Razor + unique peptides strepstr3BioIDdeltaIPHEDE2	Razor + unique peptides strepstr3BioIDFe3	Razor + unique peptides strepstr3deltaDIPHEPE5	Unique peptides strepstr3BioIDdeltaIPHEDE2	Unique peptides strepstr3BioIDFe3	Unique peptides strepstr3deltaDIPHEPE5	Sequence coverage [%]	Unique + razor sequence coverage [%]	Unique sequence coverage [%]	Mol. weight [kDa]	Sequence length	Sequence lengths	Q-value	Score	Sequence coverage strepstr3BioIDdeltaIPHEDE2 [%]	Sequence coverage strepstr3BioIDFe3 [%]	Sequence coverage strepstr3deltaDIPHEPE5 [%]	Intensity	Intensity strepstr3BioIDdeltaIPHEDE2	Intensity strepstr3BioIDFe3	Intensity strepstr3deltaDIPHEPE5	MS/MS Count	Only identified by site	Reverse	Potential contaminant	id	Peptide IDs	Peptide is razor	Mod. peptide IDs	Evidence IDs	MS/MS IDs	Best MS/MS	Oxidation (M) site IDs	Oxidation (M) site positions
```
I still have the problem, more specifically this is the error that I get from galaxy:
```
File "/home/mariano/Downloads/galaxy/database/shed_tools/toolshed.g2.bx.psu.edu/repos/bornea/saint_preprocessing/211fd8ae6edc/saint_preprocessing/SAINT_preprocessing_mq_pep.py", line 86
    print "Error: bad bait " + str(baits[i])
                           ^
SyntaxError: invalid syntax
```
My file in galaxy is of .txt data type...

**Comment by mavino** [comment]
When run with docker the file is recognised but then in pre-processing SAINT I get this error:

```
Fatal error: Unknown error
Loading required package: BiocGenerics
Loading required package: methods
Loading required package: parallel

Attaching package: 'BiocGenerics'

The following objects are masked from 'package:parallel':

    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
    clusterExport, clusterMap, parApply, parCapply, parLapply,
    parLapplyLB, parRapply, parSapply, parSapplyLB

The following objects are masked from 'package:stats':

    IQR, mad, xtabs

The following objects are masked from 'package:base':

    Filter, Find, Map, Position, Reduce, anyDuplicated, append,
    as.data.frame, as.vector, cbind, colnames, do.call, duplicated,
    eval, evalq, get, grep, grepl, intersect, is.unsorted, lapply,
    lengths, mapply, match, mget, order, paste, pmax, pmax.int, pmin,
    pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,
    tapply, union, unique, unlist, unsplit

Loading required package: Biobase
Welcome to Bioconductor

    Vignettes contain introductory material; view with
    'browseVignettes()'. To cite Bioconductor, see
    'citation("Biobase")', and for packages 'citation("pkgname")'.

Loading required package: GenomicFeatures
Loading required package: S4Vectors
Loading required package: stats4
Loading required package: IRanges

Attaching package: 'IRanges'

The following object is masked from 'package:data.table':

    shift

Loading required package: GenomeInfoDb
Loading required package: GenomicRanges
Loading required package: AnnotationDbi
Loading required package: grid
Loading required package: futile.logger
Error in as.data.frame(map_peptides_proteins(peptides_txt)) : 
  error in evaluating the argument 'x' in selecting a method for function 'as.data.frame': Error in data.frame(peptides_in[i, ], mapped_protein = protein_names_split[j],  : 
  arguments imply differing number of rows: 1, 0
Calls: map_peptides_proteins -> data.frame
Calls: main -> as.data.frame
Execution halted
Traceback (most recent call last):
  File "/shed_tools/toolshed.g2.bx.psu.edu/repos/bornea/saint_preprocessing/09b89febcf98/saint_preprocessing/SAINT_preprocessing_mq_pep.py", line 100, in <module>
    bait_create(make_bait, infile)
  File "/shed_tools/toolshed.g2.bx.psu.edu/repos/bornea/saint_preprocessing/09b89febcf98/saint_preprocessing/SAINT_preprocessing_mq_pep.py", line 70, in bait_create
    read_infile = open(infile, "r")
IOError: [Errno 2] No such file or directory: './tukeys_output.txt'
```

**Comment by pstew** [comment]
Are you still having that first issue or was it resolved? If you are still having the first issue, then I talked it over with the co-authors. Is this Galaxy running Python 2.7.x or 3? The error looks like it didn't like the print. Python 3 handles prints differently:

```
Python 2.x: print "The answer is", 2*2
Python 3: print("The answer is", 2*2)
```

That second error looks like it is having trouble parsing the peptides file and outputting the rolled up proteins file (tukeys_output.txt). Do your bait names match the peptides file? Was the peptides.txt modified in any way (and is it still tab delimited)? Is the species human?

**Comment by mavino** [comment]
I am still having both the issues...not sure of python on galaxy...I just recently downloaded galaxy locally and my computer is pretty much new (Ubuntu 18), so it should be everything in python 3...
I don't have a bait file, I was hoping the software could have done it for me...
the peptides.txt file was just modified to accommodate no spaces in the labels like you asked me before...
I was not associating a species name with the file when I upload it on galaxy, however my experiments come from Saccharomyces pombe.

**Comment by pstew** [comment]
Can you please

- Use http://apostl.moffitt.org/ to ensure all libraries, Python, etc. are properly set up. This should fix the print issue.
- Provide the FASTA file your proteomics facility used to search the data with MaxQuant to the pipeline. This is may be the culprit for the Tukey's error (I think by default it is trying to match to homo sapiens).
- Use the bait create tool first to create a bait file independent of the provided workflows. Make the bait names match exactly the `<name>` in the MaxQuant file where it the column header says `Intensity <name>`.

**Comment by mavino** [comment]
OK I am using http://apostl.moffitt.org/, I am now providing the right fasta file for my organism of interest, the same that was used in MaxQuant and I am creating the bait file following the procedure in bait create tool.
This is the error now I get:
```
Fatal error: Unknown error
Loading required package: BiocGenerics
Loading required package: parallel

Attaching package: ‘BiocGenerics’

The following objects are masked from ‘package:parallel’:

    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
    clusterExport, clusterMap, parApply, parCapply, parLapply,
    parLapplyLB, parRapply, parSapply, parSapplyLB

The following objects are masked from ‘package:stats’:

    IQR, mad, sd, var, xtabs

The following objects are masked from ‘package:base’:

    anyDuplicated, append, as.data.frame, basename, cbind, colnames,
    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
    union, unique, unsplit, which, which.max, which.min

Loading required package: Biobase
Welcome to Bioconductor

    Vignettes contain introductory material; view with
    'browseVignettes()'. To cite Bioconductor, see
    'citation("Biobase")', and for packages 'citation("pkgname")'.

Loading required package: GenomicFeatures
Loading required package: S4Vectors
Loading required package: stats4

Attaching package: ‘S4Vectors’

The following objects are masked from ‘package:data.table’:

    first, second

The following object is masked from ‘package:base’:

    expand.grid

Loading required package: IRanges

Attaching package: ‘IRanges’

The following object is masked from ‘package:data.table’:

    shift

Loading required package: GenomeInfoDb
Loading required package: GenomicRanges
Loading required package: AnnotationDbi
Loading required package: grid
Loading required package: futile.logger
Error in data.frame(peptides_in[i, ], mapped_protein = protein_names_split[j],  : 
  arguments imply differing number of rows: 1, 0
Calls: main ... as.data.frame -> map_peptides_proteins -> data.frame
Execution halted
Traceback (most recent call last):
  File "/home/bornea/shed_tools/toolshed.g2.bx.psu.edu/repos/bornea/saint_preprocessing/2d78642361c3/saint_preprocessing/SAINT_preprocessing_mq_pep.py", line 103, in <module>
    bait_temp_file = open(sys.argv[10], 'r')
IOError: [Errno 2] No such file or directory: 'None'
```

**Comment by pstew** [comment]
Seems like something is up with the bait or the peptides file. Can you please share your history and provide us with the link so we can take a look? https://galaxyproject.github.io/training-material/topics/introduction/tutorials/galaxy-intro-101/tutorial.html#share-your-work

**Comment by mavino** [comment]
http://apostl.moffitt.org/u/marianoavino/h/unnamed-history

thank you very much

**Comment by bornea** [comment]
Okay I see it now. You are using the proteinGroups file created by maxquant
rather than the peptides file from maxquant file that APOSTL uses.
Additionally in this case you have set "Are You Providing Your Own bait
file?" to Yes which will expect a bait.txt file. I am sorry it took me so
long to notice the different file being used here.

Adam Borne

**Comment by mavino** [comment]
I put  "Are You Providing Your Own bait file?" to NO (I am very sorry about that...) and I ended up again with the tukey error
```IOError: [Errno 2] No such file or directory: './tukeys_output.txt'```

However I am still using the file proteinGroups, I guess I will try with peptides.txt now (which was the first file I used at the beginning of this chat...) thinking the two could have interchangeable (sorry I am quite new in this). Thanks a lot again. I will update the situation.
